projects/plugins/jetpack/extensions/blocks/contact-form/transforms.js

```diff
@@ -0,0 +1,138 @@
+    const prefix = 'jetpack';
+
+    switch ( type ) {
+        case 'text': case 'url':
+            return `${prefix}/field-text`;
+        case 'textarea':
+            return `${prefix}/field-textarea`;
+        case 'radio':
+            return `${prefix}/field-radio`;
+        case 'checkbox':
+            return `${prefix}/field-checkbox`;
+        case 'select':
+            return `${prefix}/field-select`;
+        case 'email':
+            return `${prefix}/field-email`;
+        case 'name':
+            return `${prefix}/field-name`;
+        default:
+            return `${prefix}/field-text`;
+    }
+};
+
+const transformContactFormShortcode = ( shortcode ) => {
+    const blockAttrs = {
+        to: getAttributeValue( 'contact-form', 'to', shortcode ),
+        subject: getAttributeValue( 'contact-form', 'subject', shortcode ),
+        submitButtonText: getAttributeValue( 'contact-form', 'submit_button_text', shortcode )
+    };
+
+    return {
+        blockName: 'jetpack/contact-form',
+        attrs: pickBy( blockAttrs, identity )
+    };
+};
+
+const transformContactFieldShortcode = ( shortcode ) => {
+    const blockAttrs = {
+        label: getAttributeValue( 'contact-field', 'label', shortcode ),
+        placeholder: getAttributeValue( 'contact-field', 'placeholder', shortcode ),
+        required: getAttributeValue( 'contact-field', 'required', shortcode ),
+        options: getAttributeValue( 'contact-field', 'options', shortcode ),
+    };
+
+    const blockName = getContactFieldBlockName( getAttributeValue( 'contact-field', 'type', shortcode ) );
+
+    // Split block option values into an array.
+    if ( blockAttrs.options ) {
+        blockAttrs.options = blockAttrs.options.split( ',' );
+    }
+
+    return createBlock( blockName, pickBy( blockAttrs, identity ) );
+};
+
+const blockData = {
+    root: {},
+    innerBlocks: []
+};
+
+export default {
+    from: [
+        {
+            type: 'raw',
+            priority: 1,
+            isMatch: ( node ) => {
+                if (
+                    node.nodeName === 'P' &&
+                    ( /\[contact-form(\s.*?)?\](?:([^\[]+)?)?/g.test( node.textContent ) ||
+                    /\[contact-field(\s.*?)?\](?:([^\[]+)?)?/g.test( node.textContent ) ||
+                    /\[\/contact-form]/g.test( node.textContent ) )
+                ) {
+                    return true;
+                }
+
+                return false;
+            },
+            transform: ( node ) => {
+                const shortCode = node.textContent;
+
+                if ( shortCode.includes( '[contact-form' ) ) {
+                    blockData.root = {};
+                    blockData.innerBlocks = [];
+
+                    blockData.root = transformContactFormShortcode( shortCode );
+                }
+
+                if ( shortCode.includes( '[contact-field' ) ) {
+                    blockData.innerBlocks.push( transformContactFieldShortcode( shortCode ) );
```

Comment:
I might be missing something, but I can't see how this loops through all the nested `contact-fields`, and when I test it does only transform one of them?

Reply:
Do you have new lines after every contact field? I think that may be the issue -- not picking up each shortcode as a separate shortcode.

Reply:
I tried with or without a newline after each field and I get the same result.

Reply:
hmm, that's odd. Are you using the shortcode example from the PR description?

Reply:
It doesn't need to loop, it should be treating every contact field as a separate shortcode.

Reply:
yeh, just copying and pasting your code ... very strange ... I will do some debugging and try and work out what is going on - on the surface it appears to be just treating it as a single string, and doing a too greedy regex on it so attributes from different fields end up in the single block ... not sure why it does that for me and not you